lisa/sut_orchestrator/azure/features.py

```diff
@@ -896,6 +957,108 @@ def reload_module(self) -> None:
+        self,
+        virtual_network_name: str,
+        subnet_name: str,
+        subnet_mask: str,
+        route_table: Any,
+    ) -> bool:
+        platform: AzurePlatform = self._platform  # type: ignore
+        network_client = get_network_client(platform)
+        subnet_az = network_client.subnets.get(
+            resource_group_name=self._resource_group_name,
+            virtual_network_name=virtual_network_name,
+            subnet_name=subnet_name,
+        )
+        self._log.debug(f"Checking subnet: {subnet_az.address_prefix} == {subnet_mask}")
+        # Step 4: once we find the matching subnet, assign the routing table to it.
+        if subnet_az.address_prefix == subnet_mask:
+            subnet_az.route_table = route_table
+            result = network_client.subnets.begin_create_or_update(
+                resource_group_name=self._resource_group_name,
+                virtual_network_name=virtual_network_name,
+                subnet_name=subnet_name,
+                subnet_parameters=subnet_az,
+            ).result()
+            # log the subnets we're finding along the way...
+            self._log.info(
+                f'Assigned routing table "{route_table}" to subnet: "{subnet_az}"'
+                f' with result: "{result}"'
+            )
+            return True
+        return False
+
+    # Subroutine to create the route table,
+    # seperated because the create/apply process has multiple potential timeouts.
+    # We don't want to restart the entire process if one step fails.
+    @retry(HttpResponseError, tries=5, delay=1, backoff=1.3)
+    def _do_create_route_table(
+        self,
+        em_first_hop: str,
+        subnet_mask: str,
+        nic_name: str,
+        route_name: str,
+        next_hop_type: str,
+        dest_hop: str,
+    ) -> Union[RouteTable, None]:
+        platform: AzurePlatform = self._platform  # type: ignore
+        network_client = get_network_client(platform)
+        vm = get_vm(platform, self._node)
+
+        # Set up first hop routing rule.
+        # If no exact match first hop is provided:
+        # assume the rule is to be applied to all traffic on the subnet.
+        # Otherwise allow an arbitrary 'first hop' address
+        if not em_first_hop:
+            address_prefix = subnet_mask
+        else:
+            address_prefix = em_first_hop
+
+        # NOTE: Next Hop Types
+        # 'None' is for dropping all traffic
+        # 'VirtualAppliance' is common for sending all traffic on a subnet
+        # to a VM (or NetVirtualApplicate aka NVA ) to filter it before forarding.
+        # There are other next hop types, see:
+        # https://learn.microsoft.com/en-us/azure/virtual-network/virtual-networks-udr-overview#user-defined
+
+        # Step 1: Create the routing table entry, it will be assigned to a subnet later.
+        route_table_name = f"{nic_name}-{route_name}-route_table"
+        route_table: Union[
+            RouteTable, None
```

Comment:
Is it possible to be None, after create or update? Even so, the `result()` will fail always. So, make it simpler, the type should be only `RouterTable`.

Reply:
Ah nice catch. Changed the return type.

Reply:
Haven't see it's changed, it should be like `route_table: RouteTable = `